agent/api.go

```diff
@@ -63,7 +63,9 @@ func (a *agent) apiHandler() http.Handler {
 	r.Get("/debug/manifest", a.HandleHTTPDebugManifest)
 	r.Get("/debug/prometheus", promHandler.ServeHTTP)
 
-	return r
+	return r, func() error {
+		return containerAPI.Close()
+	}
```

Comment:
Review: We don't want to just rely on agent context, we will also explicitly clean up running services in the containerAPI.